api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -147,11 +147,18 @@ public boolean purgeComplexData(Obs obs) {
 	 * @return File object
 	 */
 	public static File getComplexDataFile(Obs obs) {
+		if (!StringUtils.isBlank(obs.getValueComplex())) {
 		String[] names = obs.getValueComplex().split("\\|");
 		String filename = names.length < 2 ? names[0] : names[names.length - 1];
 		File dir = OpenmrsUtil.getDirectoryInApplicationDataDirectory(
 		    Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR));
 		return new File(dir, filename);
+		}
+		else {
+			log.error("The file must exist!!");
```

Comment:
@HerbertYiga I think the error message should be something like `The file does not exist` since this method is used in a GET API.